src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        increaseSize();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndexPosition(index);
+        increaseSize();
+        System.arraycopy(values, index, values,
+                index + 1, size - index);
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexPosition(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexPosition(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexPosition(index);
+        T removedElement = values[index];
+        deleteElementInArray(index);
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T incomeElement;
+        for (int i = 0; i < values.length; i++) {
+            T currentElement = values[i];
+            if (currentElement == element
+                    || currentElement != null
+                    && currentElement.equals(element)) {
+                incomeElement = element;
+                deleteElementInArray(i);
+                return incomeElement;
+            }
+        }
+        throw new NoSuchElementException("");
```

Comment:
Here you can reuse more top-level method
```suggestion
        for (int i = 0; i < values.length; i++) {
            if (values[i] == element || values[i] != null && values[i].equals(element)) {
                return remove(i);
            }
        }
        throw new NoSuchElementException("..some message..");
```